fl4health/utils/losses.py

```diff
@@ -221,15 +221,16 @@ def aggregate_losses_dict(
             Dict[str, torch.Tensor]: A single dictionary with the aggregated losses according to the given loss
                 meter type
         """
-        # We don't know the keys of the dict (backward or additional losses) beforehand. We obtain them
-        # from the first entry because we know all of the losses will have the same keys
-        loss_keys = loss_list[0].keys()
-        num_losses = len(loss_list)
+        # We don't know the keys of the dict (backward or additional losses) beforehand. We don't obtain them
+        # from the first entry because losses can have different keys. We get list of all the keys from
+        # all the losses.
+        loss_keys = list(set(key for loss_dict_ in loss_list for key in loss_dict_.keys()))
```

Comment:
Do we need to convert the `set` here to a `list`? I think we can iterate through the set of loss_keys similar to the way loop through the list as it is.